fix(lsp): Ensure users get feedback on references/symbols errors or empty results

mfussenegger opened the pull request:
Relates to https://github.com/neovim/neovim/issues/15050

Users should get some indication if there was an error or an empty
result.

mfussenegger committed:
fix(lsp): Ensure users get feedback on references/symbols errors or empty results

Relates to https://github.com/neovim/neovim/issues/15050

Users should get some indication if there was an error or an empty
result.

mjlbach commented on runtime/lua/vim/lsp/handlers.lua:
```diff
@@ -191,30 +191,37 @@ M['textDocument/codeLens'] = function(...)
+--@param map_result function `((resp, bufnr) -> list)` to convert the response
+--@param entity name of the resource used in a `not found` error message
+local function response_to_qflist(map_result, entity)
+  return function(err, _, result)
```
bufnr?

mfussenegger replied:
Right 🤦
Fixed it.

mfussenegger commented:
I just was that the error handling is also inconsistent:

```
  if err then
    error("Could not execute code action: "..err.message)
  end
```

```
  if err then error(vim.inspect(err)) end
```


```
    if err then
      return err_message(tostring(err))
    end
```


Should we normalize that as well?

mjlbach commented:
Up to you, we should probably use the err_message utility function instead of vim.inspect for consistency, I assume the string conversion was added initially in response to a specific issue, but that predates me.

mfussenegger commented:
I'm a bit too trigger happy today.
Turns out the error handling within the handlers didn't work to begin with because of this:

https://github.com/neovim/neovim/blob/256570a7a628aa377f405cb55343a38d186dc770/runtime/lua/vim/lsp/handlers.lua#L456-L458

And that results on a `table: 0x.......` message - not very helpful.
I'll add a second commit that fixes it.

mfussenegger committed:
fix(lsp): Ensure human readable errors are printed

`return err_message(tostring(err))` caused errors to be printed as
`table: 0x123456789` instead of showing the error code and error
message.

This also removes some `if err` blocks that never got called because at
the end of `handlers.lua` all the handlers are wrapped with logic that
adds generic error handling.
